From 4d298d82aeaac169a08fa532b59ae531c304fb59 Mon Sep 17 00:00:00 2001
From: Ian Pedowitz <ijpedowitz@google.com>
Date: Thu, 18 Jan 2018 16:24:11 -0800
Subject: [PATCH 058/306] Fixing app compat issue b/72143978

Revert "Remove obsolete workaround."

This reverts commit 5e48241a95b843c10f96c57d8544af9d04807218.

Bug: 72143978
Bug: 24465209
Test: Tested failing case on sailfish, reverted back all CL's since
Test: 3471433 for b/24465209 and apps open
(cherry picked from commit 0c3c4d1c7d6fb210548f32a6e6ec01fd9bf90760)

Change-Id: Ic7989477ea061f4be3335f0af0ce76ba62820b94
---
 native/graphics/jni/Android.bp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/native/graphics/jni/Android.bp b/native/graphics/jni/Android.bp
index 61164e068da..0704e3545b6 100644
--- a/native/graphics/jni/Android.bp
+++ b/native/graphics/jni/Android.bp
@@ -29,6 +29,13 @@ cc_library_shared {
     shared_libs: [
         "libandroid_runtime",
     ],
+
+    arch: {
+        arm: {
+            // TODO: This is to work around b/24465209. Remove after root cause is fixed
+            ldflags: ["-Wl,--hash-style=both"],
+        },
+    },
 }
 
 // The headers module is in frameworks/native/Android.bp.
-- 
2.17.1

